lib/internal/mime.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ObjectCreate,
+  MapIteratorPrototypeNext,
+  RegExpPrototypeTest,
+  RegExpPrototypeExec,
+  StringPrototypeReplace,
+  StringPrototypeToLowerCase,
+  StringPrototypeSlice,
+  StringPrototypeCharAt,
+  SymbolReplace,
+  SymbolIterator,
+  SafeMap: Map,
+} = primordials;
+
+const NotHTTPTokenCodePoint = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
+const NotHTTPQuotedStringCodePoint = /[^\t\u0020-~\u0080-\u00FF]/g;
+NotHTTPQuotedStringCodePoint[SymbolReplace] = null;
+
+const HTTP_TAB_OR_SPACE = '\t ';
+const HTTP_WHITESPACE = ArrayPrototypeJoin(['\r\n', ...HTTP_TAB_OR_SPACE], '');
+const BEGINNING_WHITESPACE = new RegExp(
+  `^[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*`);
+const ENDING_WHITESPACE = new RegExp(
+  `[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*$`);
+
+function toASCIILower(str) {
+  return StringPrototypeReplace(str, /[A-Z]/g, (c) => StringPrototypeToLowerCase(c));
+}
+
+function parseTypeAndSubtype(str) {
+  // Skip only HTTP whitespace from start
+  str = StringPrototypeReplace(str, BEGINNING_WHITESPACE, '');
+  let position = 0;
+  // read until '/'
+  const typeMatch = execPattern(/^[^/]*/, str);
+  if (!typeMatch ||
+    typeMatch[0] === '' ||
+    testPattern(NotHTTPTokenCodePoint, typeMatch[0])) {
+    throw new TypeError('Invalid MIME type');
+  }
+  position = typeMatch[0].length;
+  // skip '/'
+  position += 1;
+  const type = toASCIILower(typeMatch[0]);
+  // read until ';'
+  const subtypeMatch = execPattern(
+    /^[^;]*/,
+    StringPrototypeSlice(str, position));
```

Comment:
Isn't this just `const subtypeMatch = str.indexOf(';', position)` (or I assume `StringPrototypeIndexOf(str, ';', position)`)?

Reply:
kind of, but this will read to end of string instead of returning -1 if it isn't found. would need some extra fiddling to handle the -1 with indexOf, neutral on using either but this seemed a little less confusing honestly.

Reply:
Well, we can always:
```js
const indexOf = (str, symbol, offset = 0, default = -1) => { .. }
// and then
indexOf(str, ';', position, str.length)
```
Or even modify `indexOfAny` proposed below.
Though I myself would even prefer the explicit fiddling with `-1` as it simplifies the code by making it more explicit.

It's just that I try to avoid using RegExp unless absolutely necessary as IMO they are usually (possibly much) less efficient and complicate both the understanding and maintenance of the code. RegExps get a lot of getting used to and are "heavier" for the reader of the code to understand, instead of just seeing "find index of ';' and move current position over it" one has to  "compile" the regex first, consider how it works and then reassure itself that the above will be a result. The streamlined code of `indexOf` + `if`/`?` allows for much less interpretation and clearly shows what it is doing which makes code easier to understand.

Anyway, this is not blocking in any way so feel free to 'resolve' with whichever way you prefer.